compiler/rustc_parse/src/parser/item.rs

```diff
@@ -724,11 +724,14 @@ impl<'a> Parser<'a> {
                         //         Some(4)
                         //     }
                         // ```
-                        && self
-                            .span_to_snippet(self.prev_token.span)
-                            .map_or(false, |snippet| snippet == "}")
+                        && self.prev_token.kind == token::CloseDelim(Delimiter::Brace)
```

Comment:
As commented below, I think we should use `span_to_snippet`.

Reply:
Updated!